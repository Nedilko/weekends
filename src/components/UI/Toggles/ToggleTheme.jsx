import PropTypes from 'prop-types'
import { MoonIcon, SunIcon } from '@heroicons/react/outline'

ToggleTheme.propTypes = {
  label: PropTypes.string,
  checked: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
}

function ToggleTheme({ label, checked, onClick }) {
  return (
    <label className="flex items-center p-2 text-xl font-thin text-slate-600 dark:text-gray-400">
      {label}
      <span
        onClick={onClick}
        className={`flex justify-between w-16 h-8 items-center flex-shrink-0 ml-4 p-1 rounded-full duration-300 ease-in-out ${
          checked ? 'bg-slate-700' : 'bg-gray-200'
        }`}
      >
        <MoonIcon
          className={`w-5 h-5 ml-1 rounded-full text-yellow-500 duration-300 ease-in-out ${
            checked ? '' : 'opacity-0 scale-0'
          }`}
        />
        <SunIcon
          className={`w-5 h-5 mr-1 rounded-full text-yellow-500 duration-300 ease-in-out ${
            checked ? 'opacity-0 scale-0' : ''
          }`}
        />
        <span
          className={`absolute flex h-7 w-7 rounded-full text-yellow-500 p-1 shadow-md duration-300 ease-in-out ${
            checked ? 'translate-x-7 bg-slate-900' : 'bg-white'
          }`}
        ></span>
      </span>
    </label>
  )
}

export default ToggleTheme
