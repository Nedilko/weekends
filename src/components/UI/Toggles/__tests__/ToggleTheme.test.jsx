import ToggleTheme from '@UI/Toggles/ToggleTheme'
import { render, screen, userEvent } from '@utils/test-utils'

describe('ToggleTheme', () => {
  const clickHandler = vi.fn()

  beforeEach(() => {
    clickHandler.mockClear()
  })

  it('should render checked', () => {
    const { conainer } = render(
      <ToggleTheme isChecked={true} onClick={clickHandler} />
    )
    expect(screen.getByRole('checkbox')).toBeChecked()
    expect(conainer).toMatchSnapshot()
  })

  it('should render unchecked', () => {
    const { conainer } = render(
      <ToggleTheme isChecked={false} onClick={clickHandler} />
    )
    expect(screen.getByRole('checkbox')).not.toBeChecked()
    expect(conainer).toMatchSnapshot()
  })

  it('should handle click', async () => {
    const user = userEvent.setup()
    render(<ToggleTheme isChecked={true} onClick={clickHandler} />)
    await user.click(screen.getByRole('checkbox'))
    expect(clickHandler).toHaveBeenCalledOnce()
  })
})
