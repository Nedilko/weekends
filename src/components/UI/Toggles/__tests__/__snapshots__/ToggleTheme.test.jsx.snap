// Vitest Snapshot v1

exports[`ToggleTheme > should render checked 1`] = `undefined`;

exports[`ToggleTheme > should render unchecked 1`] = `undefined`;

exports[`theme toggle is > in the document 1`] = `
<span
  aria-checked="true"
  class="flex h-8 w-16 flex-shrink-0 cursor-pointer items-center justify-between rounded-full bg-slate-700"
  role="checkbox"
>
  <svg
    aria-hidden="true"
    class="ml-1.5 h-5 w-5 rounded-full text-yellow-500 duration-300 ease-in-out "
    fill="none"
    stroke="currentColor"
    stroke-width="2"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
  <svg
    aria-hidden="true"
    class="mr-1.5 h-5 w-5 rounded-full text-yellow-500 duration-300 ease-in-out scale-0 opacity-0"
    fill="none"
    stroke="currentColor"
    stroke-width="2"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
  <span
    class="absolute ml-0.5 flex h-7 w-7 rounded-full text-yellow-500 shadow-md duration-300 ease-in-out translate-x-8 bg-slate-900"
  />
</span>
`;
